app/repository/sbrctrl/RestAdminDataUnitLinksEditRepository.scala

```diff
@@ -6,25 +6,27 @@ import com.typesafe.scalalogging.LazyLogging
 import repository.DataSourceNames._
 import repository._
 import repository.rest._
-import uk.gov.ons.sbr.models.{ Period, VatRef }
-import uk.gov.ons.sbr.models.edit.Patch
+import uk.gov.ons.sbr.models._
 import unitref.VatUnitRef
+import utils.edit.PatchCreation
 
-import scala.concurrent.ExecutionContext.Implicits.global
 import scala.concurrent._
 
 class RestAdminDataUnitLinksEditRepository @Inject() (@Named(SbrCtrl) unitRepository: Repository) extends AdminDataUnitLinksEditRepository with LazyLogging {
 
-  override def patchVatParentUnitLink(patch: Patch, period: Period, vatref: VatRef): Future[EditParentLinkStatus] = {
-    val path = EditAdminDataPath(period, VatUnitRef.toIdTypePair(vatref))
-    unitRepository.patchJson(path, patch).map(patchStatusToEditStatus)
+  override def updateVatParentUnitLink(from: IdAndType, to: IdAndType, vatref: VatRef, period: Period): Future[PatchStatus] = {
+    val updatePatch = PatchCreation.buildUpdateParentPatch(from, to)
+    val vatIdAndType = VatUnitRef.toIdTypePair(vatref)
```

Comment:
You can pattern match here as part of your assignment to avoid having to use _1 and _2 when constructing the UnitKey.

```val (unitId, unitType) = VatUnitRef.toIdTypePair(...)```